### ISSUE TITLE
Merge Update not working as expected with more than 1000 columns

### ISSUE THREAD

**Issue by sivasai-quartic** [delta-io/delta#479]
Is there any limit on the number of columns for merge update operation on the delta table? or any recommended number of columns?

I have a streaming job that gets 1k columns & 10 records for every batch, with that I'm trying to update the delta table already having `k columns & 100 records.
Currently, it's not even trying to execute the delta merge upsert line in my code. After the execute statement on merge update, It got to hang for more 5hr's even I have provided 64gb ram (not completed)
Below is the code sample:
```
val streamingBatchDF = spark.read.format("delta").load("file:///home/srds/delta_sink/testm_90").limit(1).toDF
val storedDeltaTablePath = "file:///home/srds/delta_sink/test_m12"
DeltaTable.forPath(spark,storedDeltaTablePath).as("sampletable").merge(streamingBatchDF.as("second_data"),"sampletable.dd = second_data.dd and sampletable.timestamp = second_data.timestamp").whenMatched().updateAll().whenNotMatched().insertAll().execute()
```
Here I'm trying merge update an only single record. Still, it's hanging. No jobs are getting generated in spark UI.
Not able to see the execution plan or what happening behind this. If the merge update with more columns is not supported on the delta table, please let us know the recommended size.

**Comment by rahulsmahadev** [comment]
HI @sivasai-quartic thanks for reporting this, we will investigate and update this issue.

**Comment by tdas** [comment]
@rahulsmahadev @sivasai-quartic Any update on this?

**Comment by nileshsinha1993** [comment]
Facing the similar issue? Any idea when this PR will be live.

**Comment by biancaberdugo** [comment]
Seems it was fixed at the version 1.2.0 https://github.com/delta-io/delta/releases/tag/v1.2.0

**Comment by nkarpov** [comment]
Thank you @biancaberdugo, this was fixed as you noted, https://github.com/databricks/runtime/pull/32405
